docs/kb/semgrep-supply-chain/scanning_multiple_lockfiles.md

```diff
@@ -0,0 +1,8 @@
+# How to scan multiple or nested lock files
+
+Semgrep Supply Chain uses lockfiles as part of its reachability analysis by determining the exact version of a dependency that a codebase is using.  Semgrep scan reliably parses lockfiles - go.mod’s, gemfile.lock’s, package-lock.json, requirement.txt, etc - to articulate all packages, libraries sourced in the compilation and subsequently report on known, associated common vulnerabilities and exposures (CVE’s).  
+
+By default, Semgrep sources any lockfiles located only  in a repo’s top-level directory.  If the repo carries multiple lockfiles, nested in other locations, you can still achieve full coverage by temporarily moving these nested lockfiles to the top-level directory, assuming they have unique names.  Semgrep will scan all lockfiles in the top-level directory of a repo. 
```

Comment:
I don't think this is the case, is it? I actually wrote a different KB article about Supply Chain troubleshooting that goes over how we find lockfiles, and Andy reviewed it and didn't comment on that: https://semgrep.dev/docs/kb/semgrep-supply-chain/why-no-findings/#is-the-supported-lockfile-present-in-an-appropriate-location-with-the-expected-name

If I got that wrong, I should definitely fix it and probably point out to this.

Reply:
@andy-r2c @mmcqd Hey y'all, could you shed some light on this please? I'm pretty sure our behavior is that SSC looks for a lockfile starting from the child directory upwards to parent? We definitely can read subdirs for lockfiles, because that's how we handle monorepos afaik. Any insight would be great :)

Reply:
I have confirmed with the team and edited the article for brevity and clarity on this matter. Resolving.

<img width="720" alt="clarification" src="https://github.com/returntocorp/semgrep-docs/assets/20766840/ffe305d8-48ec-4361-ad45-0ff5992e8f88">